### DISCUSSION TITLE
Latest Docker image 220302-0059f429 won't start

### DISCUSSION THREAD

**Discussion by bunkerman** [photoprism/photoprism#2120] [Q&A] [answered]
Hi,

I pulled the latest image 220302 and got this error

`init as root
image: amd64-prod, build 220302-bullseye
home: /photoprism
umask: "0002" (u=rwx,g=rwx,o=rx)
install-path: /opt/photoprism
running as uid 0
photoprism start
/entrypoint.sh: line 104: audit.sh: command not found`

I then downgraded to 20220121 and it started normal.

Is there something special to do upgrading to the latest image?

Regards,
Chris

**Comment by lastzero** [comment, marked as answer]
Did you manually set the PATH variable?

**Reply by bunkerman** [reply]
no i did not

**Reply by lastzero** [reply]
Any other custom changes? Kubernetes? Anything?

**Reply by lastzero** [reply]
Do you use our official examples available at dl.photoprism.app?

**Reply by bunkerman** [reply]
> Did you manually set the PATH variable?

![Screenshot_20220305-134024__01](https://user-images.githubusercontent.com/7395381/156883459-e922dc95-c452-4634-a63f-87d0e4c779c6.jpg)


do you mean this variable?

**Reply by bunkerman** [reply]
> Any other custom changes? Kubernetes? Anything?

i use a MySQL container, but it worked before

**Reply by lastzero** [reply]
The paths we use changed, so it stopped working.

**Reply by bunkerman** [reply]
ok, I deleted all the paths and it worked. thanks!

**Reply by mrbreitkreutz** [reply]
I'm getting this same error. Could you explain what you did to fix it? I tried deleting paths but still the same issue

**Reply by bunkerman** [reply]
I'm using Portainer, but I deleted all PATH related environment variables and pulled the image
![grafik](https://user-images.githubusercontent.com/7395381/156911351-b6db002e-f525-4856-95bc-44e891b4203b.png)

**Reply by tilllt** [reply]
I have the same problem, I have the docker-compose.yml from the example folder on GitHub. 

No paths manually set, what can I try?

**Reply by lastzero** [reply]
Any other changes or tools?

**Reply by mrbreitkreutz** [reply]
I deleted all the mentioned paths, it still didn’t work, then readded all the photoprism_*_paths from the docker-compose.yaml (no paths environment by itself) and this seemed to work. Not sure if there is a difference between the previous and current default?

**Reply by lastzero** [reply]
Yes, see public source code. We are on vacation currently.

**Reply by tilllt** [reply]
Edit: forget my reply , I restarted the container again and this time it worked. Very weird.

~~In my setup there are with 99% certainty no changes in the setup, except for the fact that ouroboros auto updated the container to a new version and the debian server is running unattended-upgrades which I don't constantly track.

- my docker-compose.yml is from the examples folder in GitHub, but an older version
- there are no paths defined and I didn't change or add any manually~~

**Reply by lastzero** [reply]
Maybe it's a really old version that had the assets path defined? Also note that ouroboros is not maintained anymore.

**Reply by lastzero** [reply]
Without more information, it's a lot of guesswork. We suggest you compare your config with the current example and/or sign up as a sponsor so we can help you via email.

**Reply by tilllt** [reply]
See my edit in https://github.com/photoprism/photoprism/discussions/2120#discussioncomment-2314440

There must have been a hiccup between versions. I restarted the container AGAIN and it works. Maybe there was a broken version in between? 

Btw ouroboros is unsupported but so far it seems to work so why change :)

Btw2 I am a sponsor

**Reply by lastzero** [reply]
Could be that env variables are preserved during automatic upgrades which may be a bug in ouroboros 🤔

**Reply by lastzero** [reply]
There was no broken stable version in between.

**Reply by tilllt** [reply]
Maybe I should look into a different update tool after all.

**Reply by danielo515** [reply]
So, what was the paths change for? And how does it affect the previously existing configurations?

**Reply by lastzero** [reply]
It's internal and you should never externally set a path.

**Reply by lastzero** [reply]
Check our pubic source code to see all changes and what the code does 👍

**Reply by vanderneut** [reply]
I have the same issue with a fresh install of the latest version of photoprism in Docker on a Synology NAS DS1821+, following instructions from the video at https://www.youtube.com/watch?v=t7ymSBIOJmc -- Sounds like I need to change something to get this to work with the latest version?

**Reply by vanderneut** [reply]
Backed up to older version for now as workaround, until I know how to solve this in latest version in my setup. 

Fresh install on same Synology DS1821+ of version 20220121 of PhotoPrism did not exhibit this error, with the exact same settings chosen. I made the following edits in the Advanced Settings for my install:

- In Volumes: folder "photo" (Synology Photos default) mounted as "/photoprism/originals", and folder "docker/photo-prism/config" mounted as "/photoprism/storage"
- In Port Settings, set local port as 8080
- In Environment, set a PHOTOPRISM_ADMIN_PASSWORD

Should I do any of that differently to make this work for the latest version of PhotoPrism as well?

**Reply by lastzero** [reply]
**Again, in all cases reported so far, this is because certain tools like Portainer set the PATH variable externally, which should only be set by the container internally and not externally.**

**Comment by danielo515** [comment]
Awesome, thanks for the explanation. After removing the path it worked

**Comment by spupuz** [comment]
i use portainer with stack configuration removed all env variable in stack but still ge error:

```
started 220317 as root (amd64-prod)
usermod: no changes
Problems? Our Troubleshooting Checklists help you quickly diagnose and solve them:
https://docs.photoprism.app/getting-started/troubleshooting/
file umask....: "0002" (u=rwx,g=rwx,o=rx)
home directory: /photoprism
assets path...: /opt/photoprism/assets
storage path..: /photoprism/storage
config path...: default
cache path....: default
backup path...: /photoprism/storage/backups
import path...: /photoprism/import
originals path: /photoprism/originals
switching to uid 1000:1000
/opt/photoprism/bin/photoprism start
/scripts/entrypoint.sh: line 98: /bin/gosu: No such file or directory
```

**Reply by spupuz** [reply]
i still see path variable in container env but can't remove from there.

**Reply by lastzero** [reply]
Thanks for reporting this! The gosu binary is still in the photoprism/bin directory and must be moved to /bin. Will provide a new preview image for testing soon 👌 other preview images should work, only Debian 11 is affected

**Reply by spupuz** [reply]
oh ok i'll wait for a imge then

**Reply by lastzero** [reply]
Development Preview build has been started and should be available in the next 30 minutes. Thanks for helping us with testing! :+1:
